### ISSUE TITLE
Invalid continuation byte 0xe8 in position 37.

### ISSUE THREAD

**Issue by ilgabbibo** [malwaredllc/byob#189]
### The problem

While I try to generate a client executable file for windows using the "--freeze" parameter, I get the following error:

`UnicodeDecodeError: 'utf8' codec can't decode byte 0xe8 in position 37: invalid continuation byte`
I'm running BYOB on Windows 10 using python 2.7.17.

### Full output
```
>python client.py --name prova 192.168.0.67 7890 --freeze


88                                  88
88                                  88
88                                  88
88,dPPYba,  8b       d8  ,adPPYba,  88,dPPYba,
88P'    "8a `8b     d8' a8"     "8a 88P'    "8a
88       d8  `8b   d8'  8b       d8 88       d8
88b,   ,a8"   `8b,d8'   "8a,   ,a8" 88b,   ,a8"
8Y"Ybbd8"'      Y88'     `"YbbdP"'  8Y"Ybbd8"'
                d8'
               d8'


[>] Modules
        Adding modules...  (3 modules added to client)

[>] Imports
        Adding imports... -(26 imports from 3 modules)

[>] Payload
        Uploading payload...  -(hosting payload at: http://192.168.0.67:7891///payloads/wx8.py)

[> Stager
        Uploading stager...  -(hosting stager at: http://192.168.0.67:7891///stagers/wx8.py)

[> Dropper
        Writing dropper...  (342 bytes written to prova.py)
        Compiling executable...
 52 INFO: PyInstaller: 3.6
52 INFO: Python: 2.7.17
54 INFO: Platform: Windows-10-10.0.18362
55 INFO: UPX is not available.
Traceback (most recent call last):
  File "C:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\lib\site-packages\PyInstaller\__main__.py", line 121, in <module>
    run()
  File "C:\Python27\lib\site-packages\PyInstaller\__main__.py", line 114, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\Python27\lib\site-packages\PyInstaller\__main__.py", line 65, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\Python27\lib\site-packages\PyInstaller\building\build_main.py", line 734, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "C:\Python27\lib\site-packages\PyInstaller\building\build_main.py", line 681, in build
    exec(code, spec_namespace)
  File "/rocket/byob-master/byob\prova.spec", line 2, in <module>
    block_cipher = pyi_crypto.PyiBlockCipher(key='Skw8ABQ6PNTDK4rm')
  File "C:\Python27\lib\site-packages\PyInstaller\archive\pyz_crypto.py", line 56, in __init__
    self._aesmod = import_aes(get_crypto_hiddenimports())
  File "C:\Python27\lib\site-packages\PyInstaller\archive\pyz_crypto.py", line 39, in get_crypto_hiddenimports
    import_aes(modname)
  File "C:\Python27\lib\site-packages\PyInstaller\archive\pyz_crypto.py", line 23, in import_aes
    return __import__(module_name, fromlist=[module_name.split('.')[-1]])
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\AES.py", line 50, in <module>
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\_AES.py", line 7, in <module>
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\_AES.py", line 6, in __bootstrap__
Traceback (most recent call last):
  File "client.py", line 438, in <module>
    main()
  File "client.py", line 192, in main
    dropper = _dropper(options, var=var, key=key, modules=modules, imports=imports, hidden=hidden, url=stager)
  File "client.py", line 420, in _dropper
    name = generators.freeze(name, icon=options.icon, hidden=kwargs['hidden'])
  File "D:\rocket\byob-master\byob\core\generators.py", line 246, in freeze
    util.display(line, color='reset', style='dim')
  File "D:\rocket\byob-master\byob\core\util.py", line 399, in display
    output = output.decode('utf-8')
  File "C:\Python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe8 in position 37: invalid continuation byte

```

I did not get this error while running it on Parrot OS, it happens only on windows.

**Comment by ilgabbibo** [comment]
Update: even by installing and adding UPX to the system path, nothing changes.

```
D:\byob-master\byob>python client.py --name asdrubale 192.168.1.67 7890 --freeze


88                                  88
88                                  88
88                                  88
88,dPPYba,  8b       d8  ,adPPYba,  88,dPPYba,
88P'    "8a `8b     d8' a8"     "8a 88P'    "8a
88       d8  `8b   d8'  8b       d8 88       d8
88b,   ,a8"   `8b,d8'   "8a,   ,a8" 88b,   ,a8"
8Y"Ybbd8"'      Y88'     `"YbbdP"'  8Y"Ybbd8"'
                d8'
               d8'


[>] Modules
        Adding modules...  (3 modules added to client)

[>] Imports
        Adding imports... -(26 imports from 3 modules)

[>] Payload
        Uploading payload...  -(hosting payload at: http://192.168.1.67:7891///payloads/xMA.py)

[> Stager
        Uploading stager...  -(hosting stager at: http://192.168.1.67:7891///stagers/xMA.py)

[>] Dropper
        Writing dropper...  (342 bytes written to asdrubale.py)
        Compiling executable...
 377 INFO: PyInstaller: 3.3
378 INFO: Python: 2.7.17
378 INFO: Platform: Windows-10-10.0.18362
417 INFO: UPX is available.
Traceback (most recent call last):
  File "C:\Python27\lib\runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "C:\Python27\lib\runpy.py", line 72, in _run_code
    exec code in run_globals
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\__main__.py", line 99, in <module>
    run()
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\__main__.py", line 92, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\__main__.py", line 46, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\building\build_main.py", line 791, in main
    build(specfile, kw.get('distpath'), kw.get('workpath'), kw.get('clean_build'))
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\building\build_main.py", line 737, in build
    exec(text, spec_namespace)
  File "<string>", line 2, in <module>
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\archive\pyz_crypto.py", line 54, in __init__
    self._aesmod = import_aes(get_crypto_hiddenimports())
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\archive\pyz_crypto.py", line 37, in get_crypto_hiddenimports
    import_aes(modname)
  File "C:\Python27\lib\site-packages\pyinstaller-3.3-py2.7.egg\PyInstaller\archive\pyz_crypto.py", line 21, in import_aes
    return __import__(module_name, fromlist=[module_name.split('.')[-1]])
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\AES.py", line 50, in <module>
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\_AES.py", line 7, in <module>
  File "c:\users\off4a\appdata\local\temp\easy_install-qkqjbx\pycrypto-2.6.1-py2.7-win-amd64.egg.tmp\Crypto\Cipher\_AES.py", line 6, in __bootstrap__
Traceback (most recent call last):
  File "client.py", line 438, in <module>
    main()
  File "client.py", line 192, in main
    dropper = _dropper(options, var=var, key=key, modules=modules, imports=imports, hidden=hidden, url=stager)
  File "client.py", line 420, in _dropper
    name = generators.freeze(name, icon=options.icon, hidden=kwargs['hidden'])
  File "D:\rocket\byob-master\byob\core\generators.py", line 246, in freeze
    util.display(line, color='reset', style='dim')
  File "D:\rocket\byob-master\byob\core\util.py", line 399, in display
    output = output.decode('utf-8')
  File "C:\Python27\lib\encodings\utf_8.py", line 16, in decode
    return codecs.utf_8_decode(input, errors, True)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xe8 in position 37: invalid continuation byte
```

How do I fix this?

**Comment by ilgabbibo** [comment]
Another update: I tried to build the .exe on another machine and it worked just fine.

I think the problem is within my machine and I don't know what is the source of it, but whatever.

**Comment by malwaredllc** [comment]
Just to clarify, you were able to compile the .exe on another Windows machine, and you are able to run that .exe successfully?

**Comment by ilgabbibo** [comment]
@malwaredllc no, throws the error shown in this issue: #168

**Comment by fu050409** [comment]
I got a similar decode error, but my machine running CentOS 7.x, the client is Windows 10.

**Comment by malwaredllc** [comment]
This bug should now be fixed in #191 
Let me know if the issue persists for you.
